### PR TITLE
Expose setters on WebHeaderCollection

### PR DESCRIPTION
**Pull request by danmosemsft**
Fix https://github.com/dotnet/corefx/issues/14108

@stephentoub @CIPop @davidsh

### CODE REVIEW

File: src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs [hunk 1]
```diff
@@ -38,6 +38,17 @@ public void HttpRequestHeader_Add_Success()
             Assert.Equal("Connection", w.AllKeys[0]);
         }
 
+        [Fact]
```
Comment (stephentoub) [inline src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs]: The tests should instead be added to WebHeaderCollectionTest.netstandard17.cs; I expect that's the reason for the test failures.

File: src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs [hunk 2]
```diff
@@ -203,6 +214,18 @@ public void Setter_SameHeaderTwice_Success(string firstName, string secondName)
         }
 
         [Theory]
+        public void Setter()
```
Comment (stephentoub) [inline src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs]: Is there a more descriptive name you can pick that's more consistent with the other test names?

File: src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs (hunk 1 above)
Comment (danmosemsft) [inline src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs]: doh.

**Comment by danmosemsft** [comment]
@mmitche is there any way to figure out what might have caused the timeout (https://ci.dot.net/job/dotnet_corefx/job/master/job/ubuntu14.04_release_prtest/1041/consoleText) ? The tests only seem to have been running for 15 minutes - usually a slow test causes the timeout.

@dotnet-bot test  Innerloop Ubuntu14.04 Release Build and Test  (timeout)

**Comment by mmitche** [comment]
@danmosemsft Not unless you jump on the machine.  I don't believe that Jenkins maintains a list of killed processes and command lines after job abort.  That would be good data though.

**Pull request merged**